app/src/app/search/search.component.ts

```diff
@@ -81,6 +81,7 @@ export class SearchComponent implements OnInit, OnDestroy {
    * List of search results.
    */
   public searchResultList: SearchResultList;
+  public publicIdResult = undefined;
```

Comment:
You don't have to initialize something with undefined unless you are overriding a value.